tmk_core/common/chibios/flash_stm32.c

```diff
@@ -94,30 +113,40 @@ FLASH_Status FLASH_WaitForLastOperation(uint32_t Timeout) {
 
 /**
  * @brief  Erases a specified FLASH page.
- * @param  Page_Address: The page address to be erased.
+ * @param  Page_Address: The page address or sector to be erased.
  * @retval FLASH Status: The returned value can be: FLASH_BUSY, FLASH_ERROR_PG,
  *   FLASH_ERROR_WRP, FLASH_COMPLETE or FLASH_TIMEOUT.
  */
-FLASH_Status FLASH_ErasePage(uint32_t Page_Address) {
+FLASH_Status FLASH_ErasePage(uint32_t Page) {
     FLASH_Status status = FLASH_COMPLETE;
     /* Check the parameters */
-    ASSERT(IS_FLASH_ADDRESS(Page_Address));
+    //ASSERT(IS_FLASH_ADDRESS(Page_Address));
     /* Wait for last operation to be completed */
     status = FLASH_WaitForLastOperation(EraseTimeout);
 
     if (status == FLASH_COMPLETE) {
         /* if the previous operation is completed, proceed to erase the page */
+#if defined(EEPROM_EMU_STM32F411xE)
+        FLASH->CR &= ~FLASH_CR_SNB;
+        FLASH->CR |= (Page << FLASH_CR_SNB_POS);
+        FLASH->CR |= FLASH_CR_SER;
+#else
+        FLASH->AR = Page;
         FLASH->CR |= FLASH_CR_PER;
-        FLASH->AR = Page_Address;
+#endif
```

Comment:
you swapped these two around, any reason for that? actually this reliably bricks my F072 board by erasing the first page (at 0x08000000)

To reproduce, I'm flashing with:

```
sudo dfu-util -d 0483:DF11 -a 0 -s 0x08000000:mass-erase:force -D .build/xyz_g60_via.bin
```

After a reboot, after the first `FLASH_ErasePage` is executed, the first page is gone, set to all FFs.

reference from datasheet (dm00031936), FLASH_AR must be set after setting FLASH_CR_PER:

![image](https://user-images.githubusercontent.com/1065521/96362557-24322080-10fc-11eb-9f05-7e0d54abfd41.png)

Reply:
When I added the page erase codes for STM32F4, the register setting order was misplaced.